### DISCUSSION TITLE
Android TV support

### DISCUSSION THREAD

**Discussion by ghost** [ReVanced/revanced-discussions#167] [Question] [answered]
it'll probably be hard af to make but it would be pretty sweet to have android tv support

**Comment by MulverineX** [comment]
IMO a useless effort since https://github.com/yuliskov/SmartTubeNext exists

**Reply by DraconicNEO** [reply]
SmartTubeNext isn't really a great solution since it has many problems like not being able to play DRM content along with being very glitchy in general. It's more like NewPipe than it is like Vanced or ReVanced, yes it's good but it'll never be Vanced or Revanced.

**Comment by Degl1** [comment]
Pretty sure SmartTubeNext isn't a modified version of the YouTube app, rather a client like NewPipe but for TV. So I think Android TV support would be cool.

**Comment by MulverineX** [comment]
SmartTubeNext has pretty much all the features you'd want though and more are added every day, what benefit would come from a jank android stock port?

**Reply by DraconicNEO** [reply]
You patch it to the TV version, not the stock Android version. Benefits you get is that you'll get to Play DRM content (yes some YouTube videos are DRM protected) on the TV with all the additional benefits of Revanced. Also it'll still work when YouTube changes their site since it uses an API as opposed to scraping the site like SmartTube and NewPipe.

**Comment by Degl1** [comment]
Well you'dn't patch the stock Android YT apk, you'd patch the TV version of the YT APK.

**Comment by MulverineX** [comment]
That used to exist, here's some info on why it sucks: https://github.com/yuliskov/SmartTubeNext#q-what-is-smartyoutubetv-sytv-what-is-the-difference

You might be able to find a maintained fork somewhere, but basically, this revanced suggestion would require them to patch an entirely new app (The TV app is not a "port" that google made, its another codebase), design & maintain a web app oriented modding system that isn't needed for the other apps, all while another almost identical project already exists.

**Comment by MulverineX** [comment]
I am genuinely curious what concrete reasons for this there are.

**Reply by DraconicNEO** [reply]
To get the benefits of Revanced on the Android TV Ui with things like DRM content support and RYD integration.

**Comment by Toiletmason** [comment]
@MulverineX The ability to watch the free YouTube movies.

**Reply by DraconicNEO** [reply]
That's not the only reason, the Android TV version uses a dedicated API to access YouTube, unlike SmartTubeNext which uses a scraper similar to NewPipe because of that it may stop working anytime due to changes on the site whereas the Android TV version will work for much longer since it uses the same API as stock clients it will only stop working when the stock client version stops working.

**Reply by MulverineX** [reply]
A Revanced mod for Android TV would break as often if not more often than STN due to inevitably low priority & a lack of a dedicated team, so you'd end up in the same situation. STN rarely does break/experience instability, and usually when it does, by the time it would affect you an update is already out and ready to download. If you're too impatient/lazy to install updates when you launch the app, perhaps suggest to the STN team that they separate the Innertube API layer into something that can automatically update without a restart, ala Aliucord plugins.

**Comment by Sculas** [comment]
Well, has anyone tried this before with ReVanced? Besides design changes, I don't think the actual codebase would me much different. Just use the Android TV APK and see what happens. Apply micro-support, add something like video-ads to test, and install with the usual steps (install microg, etc).

**Reply by Sculas** [reply]
Well, I just tried it, and even with some workarounds (disabling compatibility checks) it didn't work. "MicroG Support" couldn't find `settings_fragment.xml` and "Disable Video Ads" couldn't find a certain signature (these are the only 2 patches I tried to apply). Seems like YTTV is a completely different app? The APK contents are also completely different compared to mobile.

**Reply by oSumAtrIX** [reply]
As expected.

**Reply by ghost** [reply]
Seems like it's basically a browser. (makes sense since it's not updated that regularly)

**Comment by ghost** [comment]
SmartTubeNext is good though client (except for DRM). Tbh, YouTube's movie quality is not that good compared to other streaming platforms anyways.

**Reply by MulverineX** [reply]
The point is those ad-supported free movies, but as I've said this can and should be fixed on STN's side (or really you should just pirate, lazy folks)

**Comment by GODOSTROYER** [comment]
I do not know hard to do this it will be , But I will plead you to death Please Make a Android TV version.......The OG version haas tons of ads and SmartYoutubeTV is so laggy on my Bravia , not to mention I just can't seem to like the UI , the Original YT app UI is Far more mordern , and so different that my father , sister or basically anyone who's not me wont understand how to use it!  I promise to make a 100$ donation to charity if this ever comes to life ! We need a Revanced for Android TV!!!!!!

**Reply by TheJeterLP** [reply]
Not planned for now. Maybe in the future.

**Comment by TheJeterLP** [comment, marked as answer]
Currently we concentrate on implementing the Vanced features into ReVanced. So not planned as of now, maybe in the future.
